rotkehlchen/db/accounting_rules.py

```diff
@@ -336,3 +304,160 @@ def query_rules_and_serialize(
+
+    # if there is no accounting treatment check for callbacks
+    if (callback_data := callbacks.get(event.get_type_identifier())) is None:
+        return ids_processed
+
+    processed_events_num, callback = callback_data
+    if processed_events_num == 1:  # we know that this callback only processes the current event
+        return ids_processed
+
+    # count the number of events that will be processed if accounting ran
+    processed_events_num = callback(
+        pot=pot,
+        event=event,  # type: ignore[arg-type] # mypy doesn't recognize that this is an evm event
+        other_events=iter(next_events),  # type: ignore[arg-type]  # mypy doesn't recognize that this is an evm event
+    )
+
+    for _ in range(processed_events_num - 1):  # -1 because we exclude the current event here
+        _, next_event = next(events_iterator)
+        assert next_event.identifier is not None, 'We should only process events with identifiers'
+        ids_processed.append(next_event.identifier)
+
+    return ids_processed
+
+
+def query_missing_accounting_rules(
+        db: 'DBHandler',
+        accountant: 'Accountant',
+        accounting_pot: 'AccountingPot',
+        evm_accounting_aggregator: 'EVMAccountingAggregators',
+        events: Sequence[HistoryBaseEntry],
+) -> list[bool]:
+    """
+    For a list of events returns a list of the same length with boolean values where True
+    means that the event won't be affected by any accounting rule or processed in accounting
+    """
+    history_db = DBHistoryEvents(db)
+    with db.conn.read_ctx() as cursor:
+        # to know if an event will be processed or not in accounting we also need the related
+        # events since they might use callbacks or special treatments
+        related_events = history_db.get_history_events(
+            cursor=cursor,
+            filter_query=HistoryEventFilterQuery.make(
+                event_identifiers=list({event.event_identifier for event in events}),
+                order_by_rules=[('event_identifier', True), ('sequence_index', True)],
+            ),
+            has_premium=True,
+        )
+
+    query = 'SELECT COUNT(*) FROM accounting_rules WHERE (type=? AND subtype=? AND (counterparty=? OR counterparty=?))'  # noqa: E501
+    bindings = [
+        (
+            event.event_type.serialize(),
+            event.event_subtype.serialize(),
+            NO_ACCOUNTING_COUNTERPARTY if (counterparty := getattr(event, 'counterparty', None)) is None else counterparty,  # noqa: E501
+            NO_ACCOUNTING_COUNTERPARTY,  # account for rules based only on type/subtype
+        ) for event in related_events
+    ]
+
+    callbacks = evm_accounting_aggregator.get_accounting_callbacks()
+    bindings_and_events_iterator = zip(bindings, related_events)
+    processed_events: dict[int, bool] = {}  # map of all the events related to the one requested to wether they will be processed in accounting or not  # noqa: E501
```

Comment:
Why is this mapping needed? You already have the cache right?

```suggestion
    processed_events: dict[int, bool] = {}  # map of events to whether they are processed in accounting or not  # noqa: E501
```